docs/wiki/99 - Developers/Creating a Module.md

```diff
@@ -0,0 +1,171 @@
+
+The correct way to create a menu filter is to include additional classes that represent the idea of a
+row in the table displayed to a user. This differs from a row in the database itself; for example, a row in the 
+menu filter module may represent a joining of columns across many database tables.
+
+If you are creating a Menu Filter module, your class `php/my_new_module.class.inc` should extend
+the class `\DataFrameworkMenu`.
+
+[See this file for an example](../../../modules/dicom_archive/php/dicom_archive.class.inc).
+
+##### Data Instance
+You should create a class `php/mynewmodulerow.class.inc` to represent a Menu Filter row. This will
+provide basic structure for the row and include permission flags that can restrict data to users
+affiliated with a given Site or Project.
+
+
+[See this file for an example](../../../modules/dicom_archive/php/dicomarchiverow.class.inc).
+
+##### Row Provisioner.
+This file should be accompanied by a class `php/mynewmodulerowprovisioner.class.inc`. This file should
+contain the SQL statement used to query the database for data that will be consolidated into a
+Menu Filter row.
+
+[See this file for an example](../../../modules/dicom_archive/php/dicomarchiverowprovisioner.class.inc).
+
+#### Forms
+
+This type of module represents an HTML form and is used to submit data to the back-end.
+
+### JavaScript files
+
+Modern LORIS modules use ReactJS to present data. The content of these files will vary significantly
+depending on what functionality you want to provide. The best way to learn about creating JavaScript
+for a new module is by studying the code of an existing module that is similar to what you are
+trying to design.
+
+Be sure to familiarize yourself with our [ReactJS guidelines](./ReactJS Guidelines.md).
+
+### SQL Files
+
+A pull request to create a new module must contain SQL modifications in order for LORIS to detect the module
+and serve it with appropriate access control.
+
+Please make sure that all SQL changes follow our [SQL code standards](https://github.com/aces/Loris/blob/master/docs/SQLModelingStandard.md).
+
+#### Module permissions
+
+You must define new permissions for the module. These are used to limit users to specific actions with the module.
+
+Permissions are defined in the `permissions` table in LORIS and should be separated into view and edit permissions in most cases.
+
+In our example, you would create two new permissions: `my_new_module_view` and `my_new_module_edit`. Your PHP code within
+`php/my_new_module.class.inc` will define what these permissions allow a user to do using the `hasAccess()` function. 
+
+To add new permissions, append them to the list of existing permissions found in `SQL/0000-00-01-Permission.sql`. (This file is found in your LORIS root directory, not within the module subdirectory.)
+
+#### Modules table
+
+All modules must be listed in the file `SQL/0000-00-02-Modules.sql`. 
```

Comment:
I think it's better to point them to the script that manages the table. They shouldn't have to know anything about the table schema to add their module. It only needs to be added to the table in the default schema in order to show up in the menu on a default install.